ci/bin/lint.dart

```diff
@@ -245,11 +245,17 @@ void main(List<String> arguments) async {
   final ProcessPool pool = ProcessPool();
 
   await for (final WorkerJob job in pool.startWorkers(jobs)) {
-    if (job.result?.stdout.isEmpty ?? true) {
+    if (job.result?.exitCode == 0) {
       continue;
     }
-    print('❌ Failures for ${job.name}:');
-    print(job.result.stdout);
+    if (job.result == null) {
+      print('\n❗ A clang-tidy job failed to run, aborting:\n${job.exception}');
+      exitCode = 1;
+      break;
+    } else {
+      print('❌ Failures for ${job.name}:');
+      print(job.result.stdout);
+    }
     exitCode = 1;
```

Comment:
You could just set the top-level `exitCode` in `dart:io` and not call `exit` below: https://api.dart.dev/stable/2.10.2/dart-io/exitCode.html

Reply:
Good idea, done.